target/linux/ramips/dts/ApexeraWiBot.dts

```diff
@@ -0,0 +1,169 @@
+/dts-v1/;
+
+#include "mt7620a.dtsi"
+
+#include <dt-bindings/gpio/gpio.h>
+#include <dt-bindings/input/input.h>
+
+/ {
+    compatible = "apexera,wibot", "ralink,mt7620a-soc";
+    model = "Apexera WiBot";
+
+    chosen {
+        bootargs = "console=ttyS1,57600";
+    };
+
+    aliases {
+		led-boot = &led_boot;
+		led-failsafe = &led_status;
+		led-running = &led_status;
+		led-upgrade = &led_status;
+	};
+
+    gpio-leds {
+        compatible = "gpio-leds";
+
+        wan {
+            label = "wibot:blue:wan";
+            gpios = <&gpio1 5 GPIO_ACTIVE_HIGH>;
+        };
+
+        wifi {
+            label = "wibot:blue:wifi";
+            gpios = <&gpio1 4 GPIO_ACTIVE_HIGH>;
+        };
+
+        led_status: status {
+            label = "wibot:blue:status";
+            gpios = <&gpio1 3 GPIO_ACTIVE_HIGH>;
+        };
+
+        led_boot: boot {
+            label = "wibot:blue:boot";
+            gpios = <&gpio1 11 GPIO_ACTIVE_HIGH>;
+            default-state = "on";
+        };
+
+        reset {
+            label = "wibot:blue:reset";
+            gpios = <&gpio2 23 GPIO_ACTIVE_HIGH>;
+        };
+
+        ctrl3g {
+            label = "wibot:blue:ctrl3g";
+            gpios = <&gpio2 21 GPIO_ACTIVE_HIGH>;
+        };
+
+        usb0 {
+            label = "wibot:blue:usb0";
+            gpios = <&gpio2 20 GPIO_ACTIVE_HIGH>;
+            default-state = "on";
```

Comment:
I don't see a reason to lit this led by default. Copy/past leftover?

Assuming you are interested in indicating usb connections via this led, you might want to have a look at `target/linux/ramips/dts/ArcherC2-v1.dts` to see how to enable the usb led trigger. You need to include `kmod-usb-ledtrig-usbport` in the DEVICE_PACKAGES list to do so.

Reply:
if usb0 and usb1 is connected to a hub, how do we name the port, which is named as "ohci_port1 ehci_port1" in ` target/linux/ramips/dts/ArcherC2-v1.dts`.

Reply:
What kind of hub we are talking about? Do you mean something like:

```
               +-----------+
               |           |     +--------+
mt7620 usb +--->           +----->  usb0  |
               |  usb hub  |     +--------+
               |           |     +--------+
               |           +----->  usb1  |
               +-----------+     +--------+

```
If yes, I doubt it will be possible. Or at least I've never tried it.

I would have expected an additional usb controller chip.

Reply:
yes, this hub chip is GL850A USB 2.0 Low-power HUB Controller. it could expand to two usb slots, and we use all the two slots, usb0/usb1 could both be connected with usb disks, and usb0 could provide charging ability from its battery.

Reply:
I played a bit with the usbport led trigger and it is possible to address ports of an attached usb hub.

Following the example from the [linux kernel documentation](https://elixir.bootlin.com/linux/latest/source/Documentation/devicetree/bindings/usb/usb-device.txt), this should work:

```
gpio-leds {
	compatible = "gpio-leds";

	usb0 {
		label = "wibot:blue:usb0";
		gpios = <&gpio2 20 GPIO_ACTIVE_HIGH>;
		trigger-sources = <&hub_port1>;
		linux,default-trigger = "usbport";
	};

	usb1 {
		label = "wibot:blue:usb1";
		gpios = <&gpio2 22 GPIO_ACTIVE_HIGH>;
		trigger-sources = <&hub_port2>;
		linux,default-trigger = "usbport";
	};
};

&ehci {
	status = "okay";

	hub@1 {
		#address-cells = <1>;
		#size-cells = <0>;
		compatible = "usb2109,2813";
		reg = <1>;

		hub_port1: port@1 {
			reg = <1>;
			#trigger-source-cells = <0>;
		};

		hub_port2: port@2 {
			reg = <2>;
			#trigger-source-cells = <0>;
		};
	};
};
```

You need to adjust the compatible to use the usb vendor id (VID) and device id (PID) of the GL850A.

For reference the patch for the board I tested:

```
diff --git a/target/linux/ath79/dts/ar9344_tplink_tl-wdr4300.dtsi b/target/linux/ath79/dts/ar9344_tplink_tl-wdr4300.dtsi
index 7204dd0f4f..d81aed9a32 100644
--- a/target/linux/ath79/dts/ar9344_tplink_tl-wdr4300.dtsi
+++ b/target/linux/ath79/dts/ar9344_tplink_tl-wdr4300.dtsi
@@ -172,9 +172,16 @@
 		reg = <1>;
 		#trigger-source-cells = <0>;
 
-		hub_port1: port@1 {
+		hub@1 {
+			#address-cells = <1>;
+			#size-cells = <0>;
+			compatible = "usb2109,2813";
 			reg = <1>;
-			#trigger-source-cells = <0>;
+
+			hub_port1: port@1 {
+				reg = <1>;
+				#trigger-source-cells = <0>;
+			};
 		};
 
 		hub_port2: port@2 {
```

Reply:
are you working on this?